### PR TITLE
Add IRR function for calculating internal rate of return

### LINKED ISSUE #1591: Function IRR
Spec: https://support.microsoft.com/en-us/office/irr-function-64925eaa-9988-495b-b290-3ad0c163c1bc

### PR DESCRIPTION
**Pull request by sequba**
### Context

- Implement the IRR (Internal Rate of Return) function following the Excel specification
- IRR calculates the internal rate of return for a series of cash flows
- Uses Newton-Raphson method for iterative calculation
- Add translations for all 17 built-in languages
- Add comprehensive unit tests covering edge cases and error handling

**Implemented with the assist of Claude Code + Claude Opus 4.5**

### How did you test your changes?

- [x] All existing unit tests pass
- [x] New IRR function tests pass
- [x] Test argument validation (wrong number of arguments)
- [x] Test error conditions (#NUM! for non-convergence, missing positive/negative values)
- [x] Test value type (returns NUMBER_PERCENT)
- [x] Test handling of text, logical, and empty values in ranges
- [x] Test various calculation scenarios including Microsoft documentation examples
- [x] Test guess parameter behavior
- [x] Test edge cases (very small/large values, zero values, many periods)
- [x] Test different range orientations (vertical, horizontal, 2D)

### Types of changes

- [ ] Breaking change (a fix or a feature because of which an existing functionality doesn't work as expected anymore)
- [x] New feature or improvement (a non-breaking change that adds functionality)
- [ ] Bug fix (a non-breaking change that fixes an issue)
- [ ] Additional language file, or a change to an existing language file (translations)
- [ ] Change to the documentation

### Related issues:
1. Fixes #1591 
2.
3.

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to HyperFormula](https://hyperformula.handsontable.com/guide/contributing.html) and I confirm that my code follows the code style of this project.
- [ ] I have signed the [Contributor License Agreement](https://goo.gl/forms/yuutGuN0RjsikVpM2).
- [x] My change is compliant with the [OpenDocument](https://docs.oasis-open.org/office/OpenDocument/v1.3/os/part4-formula/OpenDocument-v1.3-os-part4-formula.html) standard.
- [x] My change is compatible with Microsoft Excel.
- [x] My change is compatible with Google Sheets.
- [x] I described my changes in the [CHANGELOG.md](https://github.com/handsontable/hyperformula/blob/master/CHANGELOG.md) file.
- [ ] My changes require a documentation update.
- [ ] My changes require a migration guide.

### CODE REVIEW

File: CLAUDE.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+- Ask questions to remove ambiguity and make sure you're speaking about the right thing.
+- Ask questions if you need more information to provide an accurate answer.
+- If you don't know something, simply say, "I don't know," and ask for help.
+- Present your answer in a structured way, use bullet lists, numbered lists, tables, etc.
+- When asked for specific content, start the response with the requested info immediately.
+- When answering based on context, support your claims by quoting exact fragments of available documents.
+
+## Code Style
+
+- When generating code, prefare functional approach whenever possible (in JS/TS use filter, map and reduce functions).
```
Comment (budnix) [inline CLAUDE.md]:
```suggestion
- When generating code, prefer functional approach whenever possible (in JS/TS use filter, map and reduce functions).
```

**Review by budnix — APPROVED** [review]
LGTM. Found some typos only.
